core/lib/server/pub-op.ts

```diff
@@ -0,0 +1,1313 @@
+				throw new PubOpCreateExistingError(pubToCreate.id);
+			}
+		});
+
+		return;
+	}
+
+	private async processRelations(
+		trx: Transaction<Database>,
+		operations: OperationsMap
+	): Promise<void> {
+		const relationsToCheckForOrphans = new Set<PubsId>();
+
+		for (const [pubId, op] of operations) {
+			const allOps = [
+				...op.relationsToAdd
+					.filter((r) => r.options.replaceExisting)
+					.map((r) => ({ type: "override", ...r })),
+				...op.relationsToClear.map((r) => ({ type: "clear", ...r })),
+				...op.relationsToRemove.map((r) => ({ type: "remove", ...r })),
+			] as RelationOperation[];
+
+			if (allOps.length === 0) {
+				continue;
+			}
+
+			// Find all existing relations that might be affected
+			const existingRelations = await trx
+				.selectFrom("pub_values")
+				.innerJoin("pub_fields", "pub_fields.id", "pub_values.fieldId")
+				.select(["pub_values.id", "relatedPubId", "pub_fields.slug"])
+				.where("pubId", "=", pubId)
+				.where("relatedPubId", "is not", null)
+				.where(
+					"slug",
+					"in",
+					allOps.map((op) => op.slug)
+				)
+				.$narrowType<{ relatedPubId: PubsId }>()
+				.execute();
+
+			// Determine which relations to delete
+			const relationsToDelete = existingRelations.filter((relation) => {
+				return allOps.some((relationOp) => {
+					if (relationOp.slug !== relation.slug) {
+						return false;
+					}
+
+					switch (relationOp.type) {
+						case "clear":
+							return true;
+						case "remove":
+							return relationOp.target === relation.relatedPubId;
+						case "override":
+							return true;
+					}
+				});
+			});
+
+			if (relationsToDelete.length === 0) {
+				continue;
+			}
+			// delete the relation values only
+			await deletePubValuesByValueId({
+				pubId,
+				valueIds: relationsToDelete.map((r) => r.id),
+				lastModifiedBy: this.options.lastModifiedBy,
+				trx,
+			});
+
+			// check which relations should also be removed due to being orphaned
+			const possiblyOrphanedRelations = relationsToDelete.filter((relation) => {
+				return allOps.some((relationOp) => {
+					if (relationOp.slug !== relation.slug) {
+						return false;
+					}
+
+					if (!relationOp.options?.deleteOrphaned) {
+						return false;
+					}
+
+					switch (relationOp.type) {
+						case "clear":
+							return true;
+						case "remove":
+							return relationOp.target === relation.relatedPubId;
+						case "override":
+							return true;
+					}
+				});
+			});
+
+			if (!possiblyOrphanedRelations.length) {
+				continue;
+			}
+
+			possiblyOrphanedRelations.forEach((r) => {
+				relationsToCheckForOrphans.add(r.relatedPubId);
+			});
+		}
+
+		await this.cleanupOrphanedPubs(trx, Array.from(relationsToCheckForOrphans));
+	}
+
+	/**
+	 * remove pubs that have been disconnected/their value removed,
+	 * has `deleteOrphaned` set to true for their relevant relation operation,
+	 * AND have no other relations
```

Comment:
maybe it's a better idea to have this be a configuration setting on the `PubType` (or the pubField?)
like, for some relations this could be desireable, like versions and discussions in the current arcadia community.

but for others, like `tag` or something, you would not want this.

Reply:
in the UI i think we can just force the user to manually delete all the relations first, but i think it would be good in certain places to automatically decide this (import actions)